test_scripts/API/Navigation/SendLocation/020_SendLocation_boundary_by_parameters.lua

```diff
@@ -0,0 +1,280 @@
+    local request_params = {
+        longitudeDegrees = 1.1,
+        latitudeDegrees = 1.1
+    }
+
+    request_params["address"] = {}
+    request_params.address[subParam] = value
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", request_params)
+
+    request_params.appID = commonSendLocation.getHMIAppId()
+
+    EXPECT_HMICALL("Navigation.SendLocation", request_params)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    :ValidIf (function(_,data)
+        local ErrorMessage = ""
+        local ValidationStatus = true
+        for _,k in pairs(addressParams) do
+            if data.params.address[k] and not request_params.address[k] then
+                ErrorMessage = ErrorMessage .. k .. "\n"
+                ValidationStatus = false
+            end
+        end
+
+        if ValidationStatus == false then
+            return false, "SDL sent redundant address parameters to HMI:\n" .. ErrorMessage
+        else
+            return true
+        end
+    end)
+
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function sendLocationImage(value, self)
+    local request_params = {
+        longitudeDegrees = 1.1,
+        latitudeDegrees = 1.1
+    }
+
+    request_params["locationImage"] = {value = value, imageType = "DYNAMIC"}
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", request_params)
+
+    request_params.appID = commonSendLocation.getHMIAppId()
+    local deviceID = commonSendLocation.getDeviceMAC()
+    request_params.locationImage.value = commonSendLocation.getPathToSDL() .. "storage/"
+    .. commonSendLocation.getMobileAppId(1) .. "_" .. deviceID .. "/" ..tostring(value)
+
+
+    EXPECT_HMICALL("Navigation.SendLocation", request_params)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function sendLocationTimeStamp(paramVal, boundValue, self)
+    local params = {
+        longitudeDegrees = 1.1,
+        latitudeDegrees = 1.1
+    }
+
+    params.timeStamp = {}
+    params.timeStamp[paramVal] = boundValue
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", params)
+
+    params.appID = commonSendLocation.getHMIAppId()
+
+    if not params.timeStamp["tz_hour"] then
+        params.timeStamp["tz_hour"] = 0
+    end
+    if not params.timeStamp["tz_minute"] then
+        params.timeStamp["tz_minute"] = 0
+    end
+
+    EXPECT_HMICALL("Navigation.SendLocation", params)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    :ValidIf (function(_,data)
+        local ErrorMessage = ""
+        local ValidationStatus = true
+        for i,_ in pairs(DateTimeParams) do
+            if data.params.timeStamp[DateTimeParams[i].name] and
```

Comment:
@GetmanetsIrina ```DateTimeParams[i].name``` is not defined and will be nil

Reply:
Done